Backbase/variants
# Remove environment destination and refactor environment variable pattern for value

### What does this PR do
 - Disables `'environment'` destination for custom properties
 - Modifies documentation to remove references to it.
 - Refactor environment variable values. See `docs/CUSTOM_PROPERTY.md`

### How can it be tested
 - Run `make validation` should return no error and output the unit test coverage.
 - All commands `init`, `setup` and `switch` should work as expected.
 - Custom properties whose values follow the pattern `"{{ envVars.NAME }}"` should use environment variables in `variants.gradle` and `variants_params.rb` depending on destination.
The following doesn't work for iOS, it will be implemented separately as part of Issue #87.

### Task
resolves #85 
resolves #55 

### Overall Unit Test Coverage
Tested 1058/1918 statements
Test Coverage: 55.16%

### Checklist:

- [x] I ran `make validation` with success
